ciscoise_connector.py

```diff
@@ -48,83 +55,139 @@ def __init__(self):
 
         self._base_url = None
         self._auth = None
+        self._ha_device = None
+        self._ers_auth = None
 
     def initialize(self):
 
         config = self.get_config()
 
         self._auth = HTTPBasicAuth(config[phantom.APP_JSON_USERNAME], config[phantom.APP_JSON_PASSWORD])
         ers_user = config.get("ers_user", None)
+        self._ha_device = config.get("ha_device", None)
         if ers_user is not None:
             self._ers_auth = HTTPBasicAuth(config["ers_user"], config["ers_password"])
-        self._base_url = 'https://{0}'.format(config[phantom.APP_JSON_DEVICE])
+        self._base_url = "https://{0}".format(config[phantom.APP_JSON_DEVICE])
+
+        if self._ha_device:
+            self._ha_device_url = "https://{0}".format(self._ha_device)
+            self._call_ers_api = self._ha_device_wrapper(self._call_ers_api)
+            self._call_rest_api = self._ha_device_wrapper(self._call_rest_api)
 
         return phantom.APP_SUCCESS
 
-    def _call_ers_api(self, endpoint, action_result, data=None, allow_unknown=True):
+    def _ha_device_wrapper(self, func):
+        def make_another_call(*args, **kwargs):
+            self.debug_print("Making call to primary device")
+            ret_val, ret_data = func(*args, **kwargs)
+
+            if phantom.is_fail(ret_val) and self._ha_device:
+                self.debug_print("Call to first device failed. Data returned: {}".format(ret_data))
+                self.debug_print("Making call to secondary device")
+                ret_val, ret_data = func(try_ha_device=True, *args, **kwargs)
+
+            return ret_val, ret_data
+
+        return make_another_call
+
+    def _call_ers_api(self, endpoint, action_result, data=None, allow_unknown=True, method="get", try_ha_device=False):
+        if self._ers_auth is None:
+            return action_result.set_status(phantom.APP_ERROR, "Please provide ERS credentials for this action"), None
```

Comment:
Can we move messages to constant and also update it with "ERS credentials in asset configuration are required for this action"

Reply:
Updated